drivers/gnss/quectel_lcx6g.c

```diff
@@ -0,0 +1,757 @@
+{
+	struct quectel_lcx6g_data *data = (struct quectel_lcx6g_data *)dev->data;
+
+	const struct modem_chat_config chat_config = {
+		.user_data = data,
+		.receive_buf = data->chat_receive_buf,
+		.receive_buf_size = ARRAY_SIZE(data->chat_receive_buf),
+		.delimiter = data->chat_delimiter,
+		.delimiter_size = ARRAY_SIZE(data->chat_delimiter),
+		.filter = NULL,
+		.filter_size = 0,
+		.argv = data->chat_argv,
+		.argv_size = ARRAY_SIZE(data->chat_argv),
+		.unsol_matches = unsol_matches,
+		.unsol_matches_size = ARRAY_SIZE(unsol_matches),
+		.process_timeout = K_MSEC(2),
+	};
+
+	return modem_chat_init(&data->chat, &chat_config);
+}
+
+static void quectel_lcx6g_init_dynamic_script(const struct device *dev)
+{
+	struct quectel_lcx6g_data *data = (struct quectel_lcx6g_data *)dev->data;
+
+	data->dynamic_match.match = data->dynamic_match_buf;
+	data->dynamic_match.separators = data->dynamic_separators_buf;
+	data->dynamic_match.separators_size = sizeof(data->dynamic_separators_buf);
+	data->dynamic_match.wildcards = false;
+	data->dynamic_match.partial = false;
+
+	data->dynamic_script_chat.request = data->dynamic_request_buf;
+	data->dynamic_script_chat.response_matches = &data->dynamic_match;
+	data->dynamic_script_chat.response_matches_size = 1;
+	data->dynamic_script_chat.timeout = 0;
+
+	data->dynamic_script.name = "pair";
+	data->dynamic_script.script_chats = &data->dynamic_script_chat;
+	data->dynamic_script.script_chats_size = 1;
+	data->dynamic_script.abort_matches = NULL;
+	data->dynamic_script.abort_matches_size = 0;
+	data->dynamic_script.callback = NULL;
+	data->dynamic_script.timeout = 10;
+}
+
+static int quectel_lcx6g_init(const struct device *dev)
+{
+	int ret;
+
+	ret = quectel_lcx6g_init_nmea0183_match(dev);
+	if (ret < 0) {
+		return ret;
+	}
+
+	quectel_lcx6g_init_pipe(dev);
+
+	ret = quectel_lcx6g_init_chat(dev);
+	if (ret < 0) {
+		return ret;
+	}
+
+	quectel_lcx6g_init_dynamic_script(dev);
+
+#ifdef CONFIG_PM_DEVICE
+	pm_device_init_suspended(dev);
+#else
+	ret = quectel_lcx6g_resume(dev);
+#endif
+	return ret;
+}
+
+#define LCX6G_INST_NAME(inst, name) \
+	_CONCAT(_CONCAT(_CONCAT(name, _), DT_DRV_COMPAT), inst)
+
+#define LCX6G_DEVICE(inst)								\
+	static struct quectel_lcx6g_config LCX6G_INST_NAME(inst, config) = {		\
+		.uart = DEVICE_DT_GET(DT_INST_BUS(inst)),				\
+		.pps_mode = DT_INST_STRING_UPPER_TOKEN(inst, pps_mode),			\
+		.pps_pulse_width = DT_INST_PROP(inst, pps_pulse_width),			\
+	};										\
+											\
+	static struct quectel_lcx6g_data LCX6G_INST_NAME(inst, data) = {		\
+		.chat_delimiter = {'\r', '\n'},						\
+		.dynamic_separators_buf = {',', '*'},					\
+	};										\
+											\
+	PM_DEVICE_DT_INST_DEFINE(inst, quectel_lcx6g_pm_action);			\
+											\
+	DEVICE_DT_INST_DEFINE(inst, quectel_lcx6g_init, PM_DEVICE_DT_INST_GET(inst),	\
+			 &LCX6G_INST_NAME(inst, data), &LCX6G_INST_NAME(inst, config),	\
+			 POST_KERNEL, 99, &gnss_api);
```

Comment:
I'd say bite the bullet now and add a `GNSS_INIT_PRIORITY` symbol, may save first few drivers from messing this up by using random numbers and a patch to unify them down the road

Reply:
Added GNSS_INIT_PRIORITY with a default of 80